docs/advanced.md

```diff
@@ -403,6 +403,51 @@ some of yargs' parsing features:
   }
 }
 ```
+## Midlewares
+Sometimes you want to perform operations on the args just before reaching the command.
```

Comment:
slight grammar nits:

```
Sometimes you might want to transform arguments before they reach the command handler.

For example, you perhaps you want to validate that credentials have been provided and otherwise load credentials from a file.

Middleware is simply a stack of functions, each of which is passed the the current parsed arguments, which it can in turn update by adding values, removing values, or overwriting values.
```